### PR TITLE
Add parser validation of the start function

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -300,7 +300,9 @@ Instance instantiate(const Module& module, std::vector<ImportedFunction> importe
     // Run start function if present
     if (module.startfunc)
     {
-        if (execute(instance, *module.startfunc, {}).trapped)
+        const auto funcidx = *module.startfunc;
+        assert(funcidx <= module.funcsec.size());
```
Comment (axic) [inline lib/fizzy/execute.cpp]: This is an assert as it is ensured in the parser. However probably we'll need to make this a proper error, depending on the design of parser vs. instantiate.

File: lib/fizzy/parser.cpp [hunk 2]
```diff
@@ -285,6 +285,9 @@ Module parse(bytes_view input)
             throw parser_error{"too many memory sections (at most one is allowed)"};
     }
 
+    if (module.startfunc && (*module.startfunc >= module.funcsec.size()))
```
Comment (gumb0) [inline lib/fizzy/parser.cpp]: I believe this is not correct, because it doesn't take into account the number of imported functions

**Comment by axic** [comment]
Rebased. @gumb0 do you perhaps want to take this over?

**Comment by gumb0** [comment]
Ok I can finish this.

**Comment by gumb0** [comment]
Ready for review @chfast @axic

**Comment by gumb0** [comment]
Rebased.

**Comment by axic** [comment]
Looks good to me (but can't approve).

**Pull request merged**